src/js/page_sections/Menu/Submenu.js

```diff
@@ -1,66 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { FormattedMessage } from 'react-intl';
-import { defineMessages, injectIntl } from 'react-intl';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import { find } from 'lodash';
+import { SUPPORTED_LANGUAGES } from 'js/i18n/constants';
 
 import SubmenuItem from 'js/page_sections/Menu/SubmenuItem';
 import ExternalLink from 'js/modules/ExternalLink';
 import I18nNavLink from 'js/modules/I18nNavLink';
 import ArrowRightSVG from 'js/svg/ArrowRight';
 
-const getSubmenuClassnames = (intl, id, openSection) => {
-  const base = 'coa-Submenu';
-  const arabicRightMenuItems = intl.locale === 'ar' && id < 5;
-  const nonArabicRightMenuItems = intl.locale !== 'ar' && id > 4;
-  const shouldAlignRight = arabicRightMenuItems ||  nonArabicRightMenuItems;
-  const alignRight = shouldAlignRight ? 'coa-Submenu--align-right' : '';
-  const open = openSection === id ? 'coa-Submenu--open' : '';
+const SUBMENU_THRESHOLD_ALIGNRIGHT_RTL = 5;
+const SUBMENU_THRESHOLD_ALIGNRIGHT_LTR = 4;
 
-  return `${base} ${alignRight} ${open}`;
+const isAlignedRight = (direction, id) => {
+  return (direction === 'rtl') ? id < SUBMENU_THRESHOLD_ALIGNRIGHT_RTL : id > SUBMENU_THRESHOLD_ALIGNRIGHT_LTR;
+};
+
+const Submenu = ({id, theme, isSubmenuOpen, handleToggleAllMenus, intl}) => {
+
+  const langObj = find(SUPPORTED_LANGUAGES, {'code': intl.locale});
+  return (
+    <ul className={`coa-Submenu
+      ${isAlignedRight(langObj.direction, id) ? 'coa-Submenu--align-right' : ''}
+      ${isSubmenuOpen ? 'coa-Submenu--open' : ''}`
```

Comment:
As discussed, I prefer keeping classNames free of lots of logic. I normally do this with a separate function like `getSubmenuClassnames()`. You've mentioned that you like the way the 'classnames' npm package works. Maybe we add that as a task for refactoring? https://www.npmjs.com/package/classnames